File: 3BIN/BINV3140-1 Programmation - question spéciales/BINV3140-A .NET/Séance 1-20200919/2. Exercices/BDActeurs/BDActeurs/src/be/ipl/film_database/domaine/MetteurEnScene.java
package be.ipl.film_database.domaine;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.Calendar;
import java.util.Iterator;
import java.util.List;

public class MetteurEnScene extends Personne implements Serializable {

	/**
	 * 
	 */
	private static final long serialVersionUID = 5952964360274024205L;
	private List<Film> filmsDirigés;
	
	public MetteurEnScene(String nom, String prenom, Calendar dateDeNaissance) {
		super(nom, prenom, dateDeNaissance);
		filmsDirigés = new ArrayList<Film>();
	}

	public boolean ajouterFilm(Film film){
		
		if (filmsDirigés.contains(film))
			return false;
		
		if (film.getMetteurEnScene() == null)
			film.setMetteurEnScene(this);
		
		filmsDirigés.add(film);
		
		
		
		return true;
		
	}
	
	public Iterator<Film> films(){
		return filmsDirigés.iterator();
	}
	
	
}

File: 3BIN/BINV3140-1 Programmation - question spéciales/BINV3140-A .NET/Séance 1-20200919/2. Exercices/Week_1/Week_1/MetteurEnScene.cs
using System;
using System.Collections.Generic;

namespace be.ipl.film_database.domaine
{   
    [Serializable]
    class MetteurEnScene : Personne
    {
		private List<Film> _filmsDirigés;

		public MetteurEnScene(String nom, String prenom, DateTime dateDeNaissance) : base(nom, prenom, dateDeNaissance)
		{
			_filmsDirigés = new List<Film>();
		}

		public bool AjouterFilm(Film film)
		{

			if (_filmsDirigés.Contains(film))
				return false;

			if (film.MetteurEnScene == null)
				film.MetteurEnScene = this;

			_filmsDirigés.Add(film);



			return true;

		}

		public IEnumerator<Film> films()
		{
			return _filmsDirigés.GetEnumerator();
		}

	}
}
